[Bug]: Panic upon creation of aws_config_config_rule resources

### Terraform Core Version

1.5.1

### AWS Provider Version

5.8.0

### Affected Resource(s)

aws_config_config_rule

### Expected Behavior

Config rule resources are deployed, as per configuration (and plan).

### Actual Behavior

Terraform plan is generated correctly, and does not indicate any potential issues prior to apply. During apply, panics and fails to deploy resource of mentioned type.

### Relevant Error/Panic Output Snippet

```shell
Acquiring state lock. This may take a few moments...
module.config.aws_config_config_rule.config_rule["EC2_EBS_ENCRYPTION_BY_DEFAULT"]: Creating...
module.config.aws_config_config_rule.config_rule["EFS_ENCRYPTED_CHECK"]: Creating...
module.config.aws_config_config_rule.config_rule["KMS_CMK_NOT_SCHEDULED_FOR_DELETION"]: Creating...
module.config.aws_config_config_rule.config_rule["S3_DEFAULT_ENCRYPTION_KMS"]: Creating...

╷
│ Error: Plugin did not respond
│ 
│   with module.config.aws_config_config_rule.config_rule["EC2_EBS_ENCRYPTION_BY_DEFAULT"],
│   on modules/monitoring/config/main.tf line 19, in resource "aws_config_config_rule" "config_rule":
│   19: resource "aws_config_config_rule" "config_rule" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.config.aws_config_config_rule.config_rule["KMS_CMK_NOT_SCHEDULED_FOR_DELETION"],
│   on modules/monitoring/config/main.tf line 19, in resource "aws_config_config_rule" "config_rule":
│   19: resource "aws_config_config_rule" "config_rule" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ 
│   with module.config.aws_config_config_rule.config_rule["S3_DEFAULT_ENCRYPTION_KMS"],
│   on modules/monitoring/config/main.tf line 19, in resource "aws_config_config_rule" "config_rule":
│   19: resource "aws_config_config_rule" "config_rule" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.config.aws_config_config_rule.config_rule["EFS_ENCRYPTED_CHECK"],
│   on modules/monitoring/config/main.tf line 19, in resource "aws_config_config_rule" "config_rule":
│   19: resource "aws_config_config_rule" "config_rule" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-aws_v5.8.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x99f42b8]

goroutine 392 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/configservice.resourceConfigRuleRead({0xf387580, 0xc001bae870}, 0xc0001fc500, {0xe06f840?, 0xc002334270?})
	github.com/hashicorp/terraform-provider-aws/internal/service/configservice/config_rule.go:255 +0x258
github.com/hashicorp/terraform-provider-aws/internal/service/configservice.resourceRulePutConfig({0xf387580?, 0xc001bae870}, 0x0?, {0xe06f840?, 0xc002334270})
	github.com/hashicorp/terraform-provider-aws/internal/service/configservice/config_rule.go:240 +0x705
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0xe06f840?, 0xc002334270?})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:111 +0x34b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xf387580?, {0xf387580?, 0xc001b61a70?}, 0xd?, {0xe06f840?, 0xc002334270?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:728 +0x87
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000ddcee0, {0xf387580, 0xc001b61a70}, 0xc001b4dad0, 0xc0001fc280, {0xe06f840, 0xc002334270})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:864 +0xa7e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0040712d8, {0xf387580?, 0xc001b61950?}, 0xc000354460)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:1024 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc00230f3e0, 0xc004bb74c0, 0xc00230f410, 0xc00230f440, {0xc00231e7c0, 0x2, 0x2}}, {0xf387580?, 0xc001b61620?}, 0xc000354460)
	github.com/hashicorp/terraform-plugin-mux@v0.11.1/tf5muxserver/mux_server_ApplyResourceChange.go:30 +0x139
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00231a5a0, {0xf387580?, 0xc001b60e10?}, 0xc0017fe0e0)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/tf5server/server.go:821 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xdd4fb40?, 0xc00231a5a0}, {0xf387580, 0xc001b60e10}, 0xc0017fe000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:422 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000986000, {0xf3943a0, 0xc0049b09c0}, 0xc001b6ed80, 0xc0023d1260, 0x162634e0, 0x0)
	google.golang.org/grpc@v1.56.1/server.go:1337 +0xdf3
google.golang.org/grpc.(*Server).handleStream(0xc000986000, {0xf3943a0, 0xc0049b09c0}, 0xc001b6ed80, 0x0)
	google.golang.org/grpc@v1.56.1/server.go:1714 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.56.1/server.go:959 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.56.1/server.go:957 +0x18c

Error: The terraform-provider-aws_v5.8.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Terraform Configuration Files

```
locals {
  rules = {
    EC2_EBS_ENCRYPTION_BY_DEFAULT = {}
    EFS_ENCRYPTED_CHECK           = {}
    KMS_CMK_NOT_SCHEDULED_FOR_DELETION = {
      input_parameters = {
        kmsKeyIds = join(",", values(var.kms_key_ids)) // e.g. var.kms_key_ids = {"ebs": "<ebs-key-id>", "s3": "<s3-key-id>"}
      }
    }
    S3_DEFAULT_ENCRYPTION_KMS = {
      input_parameters = {
        kmsKeyArns = var.kms_key_arns["s3"] // e.g. var.kms_key_arns = {"ebs": "<ebs-key-arn>", "s3": "<s3-key-arn>"}
      }
      maximum_execution_frequency = null // Not supported for rule, only triggers on config change
    }
  }
}

resource "aws_config_config_rule" "config_rule" {
  for_each = local.rules

  name = "config-rule-${replace(lower(each.key), "_", "-")}"

  input_parameters            = contains(keys(each.value), "input_parameters") ? jsonencode(each.value.input_parameters) : null
  maximum_execution_frequency = contains(keys(each.value), "maximum_execution_frequency") ? each.value.maximum_execution_frequency : "One_Hour"
  tags = {
    myTag = "true"
  }

  scope {
    tag_key   = "myTag"
    tag_value = "true"
  }

  source {
    owner             = "AWS"
    source_identifier = each.key
  }
}

```

### Steps to Reproduce

Create two KMS CMKs (any old keys, no specific configurations are required).

Run `terraform apply`, passing in key ARNs and IDs as variables:
- `var.kms_key_arns = {"ebs": "<ebs-key-arn>", "s3": "<s3-key-arn>"}`
- `var.kms_key_ids = {"ebs": "<ebs-key-id>", "s3": "<s3-key-id>"}`

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

Execution output is from logs provided by an automated pipeline, not a component that I directly manage.

I've been able to deploy manually, directly to the environment and with the same code, using the MacOS Terraform package installed via HomeBrew. This leads me to believe that it isn't a problem with my configuration itself, which seems correctly defined.

### References

Had hoped that [this](https://github.com/hashicorp/terraform-provider-aws/pull/32439) would resolve the issue, as seemed quite similar in nature, but did not.

### Would you like to implement a fix?

None

--- a/.changelog/32520.txt
+++ b/.changelog/32520.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_config_config_rule: Prevent crash from unhandled read error
+```

--- a/internal/service/configservice/config_rule.go
+++ b/internal/service/configservice/config_rule.go
@@ -189,6 +189,10 @@ func ResourceConfigRule() *schema.Resource {
 	}
 }
 
+const (
+	ResNameConfigRule = "Config Rule"
+)
+
 func resourceRulePutConfig(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
 	conn := meta.(*conns.AWSClient).ConfigServiceConn(ctx)
@@ -250,6 +254,9 @@ func resourceConfigRuleRead(ctx context.Context, d *schema.ResourceData, meta in
 		log.Printf("[WARN] ConfigService Config Rule (%s) not found, removing from state", d.Id())
 		d.SetId("")
 		return diags
+	}
+	if err != nil {
+		return append(diags, create.DiagError(names.ConfigService, create.ErrActionReading, ResNameConfigRule, d.Id(), err)...)
 	}
 
 	arn := aws.StringValue(rule.ConfigRuleArn)

--- a/internal/service/configservice/config_rule.go
+++ b/internal/service/configservice/config_rule.go
@@ -20,7 +20,6 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
 	"github.com/hashicorp/terraform-provider-aws/internal/create"
-	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
 	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
 	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
 	"github.com/hashicorp/terraform-provider-aws/internal/verify"
@@ -236,7 +235,7 @@ func resourceRulePutConfig(ctx context.Context, d *schema.ResourceData, meta int
 		_, err = conn.PutConfigRuleWithContext(ctx, &input)
 	}
 	if err != nil {
-		return sdkdiag.AppendErrorf(diags, "creating AWSConfig rule: %s", err)
+		return append(diags, create.DiagError(names.ConfigService, create.ErrActionUpdating, ResNameConfigRule, name, err)...)
 	}
 
 	d.SetId(name)
@@ -300,11 +299,11 @@ func resourceConfigRuleDelete(ctx context.Context, d *schema.ResourceData, meta 
 		_, err = conn.DeleteConfigRuleWithContext(ctx, input)
 	}
 	if err != nil {
-		return sdkdiag.AppendErrorf(diags, "Deleting Config Rule failed: %s", err)
+		return append(diags, create.DiagError(names.ConfigService, create.ErrActionDeleting, ResNameConfigRule, d.Id(), err)...)
 	}
 
 	if _, err := waitRuleDeleted(ctx, conn, d.Id()); err != nil {
-		return sdkdiag.AppendErrorf(diags, "waiting for Config Service Rule (%s) to deleted: %s", d.Id(), err)
+		return append(diags, create.DiagError(names.ConfigService, create.ErrActionWaitingForDeletion, ResNameConfigRule, d.Id(), err)...)
 	}
 
 	return diags

